src/modules/commander/commander.cpp

```diff
@@ -3988,6 +3988,13 @@ set_control_mode()
 		control_mode.flag_control_altitude_enabled = (!offboard_control_mode.ignore_velocity ||
 			!offboard_control_mode.ignore_position) && !control_mode.flag_control_acceleration_enabled;
 
+		// These are all required for altitude control
+		control_mode.flag_control_altitude_enabled	= !offboard_control_mode.ignore_alt_hold || control_mode.flag_control_altitude_enabled;
+		control_mode.flag_control_attitude_enabled	= !offboard_control_mode.ignore_alt_hold || control_mode.flag_control_attitude_enabled;
+		control_mode.flag_control_rates_enabled		= !offboard_control_mode.ignore_alt_hold || control_mode.flag_control_rates_enabled;
+		control_mode.flag_control_climb_rate_enabled= !offboard_control_mode.ignore_alt_hold || control_mode.flag_control_climb_rate_enabled;
+		control_mode.flag_control_velocity_enabled	= !offboard_control_mode.ignore_alt_hold || control_mode.flag_control_velocity_enabled;
```

Comment:
Why is `control_mode.flag_control_velocity_enabled` depending on `offboard_control_mode.ignore_alt_hold`? Velocity control refers to controlling x and y velocities but not z velocity which should be captured by `control_mode.flag_control_climb_rate_enabled`.

I would think that `control_mode.flag_control_velocity_enabled` is false when you are controlling altitude but not position or x/y velocity.

Reply:
I have turned them off one by one, and for each the altitude control fails. Only when all these are turned on it seems that the altitude control works.

Reply:
Ok, I think we need to investigate the velocity one. This seems wrong to me.

Reply:
@xgerrmann do you have time to look into the position controller and why this is required?

Reply:
Yes, I will look into this. I'll get back to this in approx 1 week.

Reply:
Ok thx.

Reply:
I found the problem. in the mc_att_controller, the attitude setpoints were only published when either position control, velocity control or acceleration control were running. I added now a condition for when altitude control is running. Now in the commander the velocity control does not need to be set anymore.